projects/hip-tests/catch/unit/memory/hipMemAdvise_v2.cc

```diff
@@ -0,0 +1,278 @@
+ *  - 2) With Location type Host
+ * Test source
+ * ------------------------
+ *  - unit/memory/hipMemAdvise_v2.cc
+ * Test requirements
+ * ------------------------
+ *  - HIP_VERSION >= 7.1
+ */
+TEST_CASE("Unit_hipMemAdvise_v2_Device_Host") {
+  auto supportedDevices = getSupportedDevices();
+  if (supportedDevices.empty()) {
+    HipTest::HIP_SKIP_TEST(
+        "Test need at least one device with managed memory support");
+  }
+
+  HIP_CHECK(hipSetDevice(supportedDevices[0]));
+
+  constexpr int N = 1024;
+  constexpr int Nbytes = N * sizeof(int);
+  constexpr int value = 10;
+  int *memPtr = nullptr;
+
+  HIP_CHECK(hipMallocManaged(reinterpret_cast<void **>(&memPtr), Nbytes,
+                             hipMemAttachGlobal));
+  REQUIRE(memPtr != nullptr);
+  std::fill_n(memPtr, N, value);
+
+  SECTION("With Device") {
+    for (int deviceId : supportedDevices) {
+      HIP_CHECK(hipSetDevice(deviceId));
+
+      hipMemLocation location;
+      location.type = hipMemLocationTypeDevice;
+      location.id = deviceId;
+
+      HIP_CHECK(
+          hipMemAdvise_v2(memPtr, Nbytes, hipMemAdviseSetReadMostly, location));
+      int *devArr = nullptr;
+      HIP_CHECK(hipMalloc(&devArr, Nbytes));
+      REQUIRE(devArr != nullptr);
+
+      copyDataKernel<<<1, N>>>(devArr, memPtr);
+
+      int hostArr[N];
+      HIP_CHECK(hipMemcpy(hostArr, devArr, Nbytes, hipMemcpyDeviceToHost));
+      HIP_CHECK(hipDeviceSynchronize());
+
+      for (int i = 0; i < N; i++) {
+        INFO("For Device " << deviceId << " At index " << i
+                           << " Expected value = " << value
+                           << " Got value = " << hostArr[i]);
+        REQUIRE(hostArr[i] == value);
+      }
+
+      HIP_CHECK(hipFree(devArr));
+    }
+  }
+
+  SECTION("With Host") {
+    hipMemLocation location;
+    location.type = hipMemLocationTypeHost;
+
+    HIP_CHECK(
+        hipMemAdvise_v2(memPtr, Nbytes, hipMemAdviseSetReadMostly, location));
+
+    for (int i = 0; i < N; i++) {
+      INFO("At index " << i << " Expected value = " << value
+                       << " Got value = " << memPtr[i]);
+      REQUIRE(memPtr[i] == value);
+    }
+  }
+
+  HIP_CHECK(hipFree(memPtr));
+}
+
+/**
+ * Test Description
+ * ------------------------
+ *  - This test case checks the following scenarios
+ *  - 1) With Location type Host Numa
+ *  - 2) With Location type Host Numa Current
+ * Test source
+ * ------------------------
+ *  - unit/memory/hipMemAdvise_v2.cc
+ * Test requirements
+ * ------------------------
+ *  - HIP_VERSION >= 7.1
+ */
+#if __linux__
+TEST_CASE("Unit_hipMemAdvise_v2_HostNuma_HostNumaCurrent") {
+  auto supportedDevices = getSupportedDevices();
+  if (supportedDevices.empty() || numa_available() < 0) {
+    HipTest::HIP_SKIP_TEST("Skipping as System does not have managed memory "
+                           "supported devices or No Numa nodes in system");
+  }
+
+  HIP_CHECK(hipSetDevice(supportedDevices[0]));
```

Comment:
same comment here, after the test is done, reset it to original device (which will be 0 i guess)